video/out/placebo/ra_pl.c

```diff
@@ -450,9 +450,11 @@ static struct ra_renderpass *renderpass_create_pl(struct ra *ra,
         .glsl_shader = params->type == RA_RENDERPASS_TYPE_COMPUTE
                             ? params->compute_shader
                             : params->frag_shader,
-        .cached_program = params->cached_program.start,
-        .cached_program_len = params->cached_program.len,
     };
+    AV_NOWARN_DEPRECATED(
```

Comment:
In my opinion, it's pretty weird to enable `-werror` and then immediately do stuff to hide the deprecations so it's not caught by it. Also this would hide deprecation messages when we compile mpv ourselves so not really a fan tbh.

Reply:
That's exactly my problem I want to avoid. Having wall of deprecation warnings that are intended to be there for the time being, until say next libplacebo required version bump.

Those warnings are shadowing useful warnings, and are not really useful in day to day development. And for the code cleaning from those "intended" deprecation one can search for `AV_NOWARN_DEPRECATED`.

Having Werror on CI have big advantage of checking warnings on other platforms, because for the most part locally we compile only on one target. And manually checking logs is not fun.

I don't push for those changes, I can keep them on my local branch, no biggie.

Reply:
I mean sure it's quite a few messages but I feel that it's easier to forget about them if you hide it in the code and have to remember to grep for them later. Then again, it's not like I'm the one that goes in and bumps the libplacebo code.

One other disadvantage of `-werror` that I can think of this that the CI would break whenever something gets deprecated. That's not something that's rare in libplacebo and it can also happen in other dependencies like ffmpeg from time to time. Not like it's a huge deal, but that's just creating more manual work for ourselves in the future when the situation inevitably occurs.

Reply:
> One other disadvantage of -werror that I can think of this that the CI would break whenever something gets deprecated.

Just add `-Wno-depreciated` to the CI as well. Aside from this (and maybe a couple other warnings??) I think having -Werror on CI is a plus instead of having warnings go unnoticed.

Reply:
If you do that you just hide all the deprecation messages... I don't really get the point. Are deprecated messages so terrible?